r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -415,3 +412,154 @@ test_that("mutate and write_dataset", {
+  gf <- group_by(df, x)
+
+  expect_dplyr_equal(input %>% mutate(y = NULL) %>% collect(), df)
+  expect_dplyr_equal(input %>% mutate(y = NULL) %>% collect(), gf)
+
+  # even if it doesn't exist
+  expect_dplyr_equal(input %>% mutate(z = NULL) %>% collect(), df)
+  # or was just created
+  expect_dplyr_equal(input %>% mutate(z = rep(1, nrow(input)), z = NULL) %>% collect(), df)
+
+  # regression test for https://github.com/tidyverse/dplyr/issues/4974
+  expect_dplyr_equal(
+    input %>% mutate(z = 1, x = NULL, y = NULL) %>% collect(),
+    data.frame(x = 1, y = 1)
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L71-L75
+# test_that("assignments don't overwrite variables (dplyr #315)", {
+#   expect_dplyr_equal(
+#     tibble(x = 1, y = 2) %>% mutate(z = {x <- 10; x}) %>% collect(),
+#     tibble(x = 1, y = 2, z = 10)
+#   )
+# })
+# NOT SURE ABOUT THIS!
+test_that("assignments don't overwrite variables (dplyr #315)", {
+  expect_dplyr_equal(
+    input %>% mutate(z = {x <- 10; x}) %>% collect(),
+    tibble(x = 1, y = 2, z = 10)
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L77-L81
+# NOT SURE ABOUT THIS!
+# test_that("can mutate a data frame with zero columns and `NULL` column names", {
+#   df <- vctrs::new_data_frame(n = 2L)
+#   colnames(df) <- NULL
+#   expect_dplyr_equal(
+#     input %>% mutate(x = 1) %>% collect(),
+#     df
+#   )
+# })
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L95-L100
+# glue is a dependency of tidyselect
+test_that("glue() is supported", {
+  expect_dplyr_equal(
+    input %>% mutate(y = glue::glue("")) %>% collect(),
+    tibble(x = 1, y = glue::glue(""))
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L102-L106
+test_that("mutate disambiguates NA and NaN (#1448)", {
+  expect_dplyr_equal(
+    input %>% mutate(y = x * 1) %>% select(y) %>% collect(),
+    tibble(x = c(1, NA, NaN))
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L102-L106
+# this is somewhat "contained" in the previous test
+# test_that("mutate handles data frame columns", {
+#   expect_dplyr_equal(
+#     input %>% mutate(new_col = data.frame(x = 1:3)) %>% select(new_col) %>% collect(),
+#     data.frame(x = 1:3)
+#   )
+#
+#   # mutate() on grouped data not supported in Arrow; this will be pulling data into R
+#   # expect_dplyr_equal(
+#   #   input %>%
+#   #     group_by(x) %>%
+#   #     mutate(new_col = x) %>%
+#   #     ungroup() %>%
+#   #     select(new_col) %>%
+#   #     collect(),
+#   #   data.frame(x = 1:3)
+#   # )
+#
+#   # ROWWISE IS NOT IMPLEMENTED
+#   # expect_dplyr_equal(
+#   #   input %>%
+#   #     rowwise(x) %>%
+#   #     mutate(new_col = x) %>%
+#   #     ungroup() %>%
+#   #     select(new_col) %>%
+#   #     collect(),
+#   #   data.frame(x = 1:3)
+#   # )
```

Comment:
Instead of trying to adapt these two expectations to succeed in the arrow package, I would recommend writing them in a way that current fails, and adding a `skip()` before them. The first expectation succeeds so you can keep that as is.